uswds/uswds-site
# USWDS-Site: Create opt-out setting for in-page navigation

# Summary
Created an opt-out setting that allows us to remove the in-page navigation component on a page-by-page basis. 

To remove the in-page navigation component, add `in_page_nav: false` to the page's front matter.

:warning: Build errors should be resolved once https://github.com/uswds/uswds-site/pull/2159 is merged.

## Related issue

Closes #2010. Closes #1931.

## Preview links
Examples of affected pages:
- [Component page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-turn-off-inpage-nav/components/overview/)
- [Performance page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-turn-off-inpage-nav/performance/)
- [Settings page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-turn-off-inpage-nav/documentation/settings/)

## Problem statement
On certain pages, the in-page navigation can be unnecessary, redundant, or so long that it extends beyond the viewport. In these instances, we should be able to remove the in-page navigation from the page. 

## Solution
This PR creates a setting in the front matter that allows us to opt-out of the in-page navigation on a page-by-page basis. The in-page navigation is still set to be included by default. 

The component was removed if any of the following conditions were met:
- The in-page navigation is empty (Example: The [performance page](https://designsystem.digital.gov/performance/))
- The in-page navigation was unnecessary/redundant (Example: the [components page](https://designsystem.digital.gov/components/))
- The in-page navigation extended beyond the viewport with a height of 1000px.  (Example: the [settings page](https://designsystem.digital.gov/documentation/settings/))
    - Note: This height value is arbitrary and can be adjusted if desired.

## Testing and review
- Confirm that the chosen opt-out method makes sense and follows convention.
- Confirm that all pages that should not have an in-page nav have been identified and the in-page navigation has been removed.
- Confirm that there are no visual regressions on these and other pages.

## Possible future efforts
### Evaluate side navigation content
We should evaluate how the side navigation and in-page navigation can work together. Now that we have in-page navigation, are there any changes we should make to the content in the side navigation? What happens if there are redundancies between the two? 

### Create option to choose the heading level that in-page navigation displays
It would be helpful to be able to control which headers the in-page navigation is pulling from `<main>`. When the in-page navigation gets very lengthy, it would be helpful to have the option to exclude H3's.
